src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.basic
+
+/-!
+# Lexicographic order on finsupps
+
+This file defines the lexicographic order on finsupps.
+-/
+
+lemma ne.or_ne_of_ne {α} {a b : α} (c : α) (ab : a ≠ b) :
+  a ≠ c ∨ b ≠ c :=
+@dite _ (a = c) (classical.prop_decidable (a = c))
+  (λ ac, or.inr (ne_of_ne_of_eq ab.symm ac)) (λ ac, or.inl ac)
+
+namespace finsupp
+
+variables {α N : Type*} [has_zero N]
+
+open_locale classical
+
+lemma filter_ne_eq_empty_iff {f g : α →₀ N} :
+  (f.support ∪ g.support).filter (λ a, f a ≠ g a) = ∅ ↔ f = g :=
+begin
+  refine ⟨λ de, _, λ fg, _⟩,
+  { ext a,
+    refine not_ne_iff.mp (λ h, not_ne_iff.mpr de _),
+    refine finset.ne_empty_of_mem (_ : a ∈ _),
+    simp only [ne.def, finset.mem_filter, finset.mem_union, finsupp.mem_support_iff],
+    exact ⟨h.or_ne_of_ne _, h⟩ },
+  { subst fg,
+    simp only [ne.def, eq_self_iff_true, not_true, finset.filter_false] }
+end
+
+variables [linear_order α]
+
+/--  The partial order on `finsupp`s obtained by the lexicographic ordering.
+`finsupp.lex` is the proof that this partial order is in fact linear. -/
+def _root_.finsupp_partial_order [partial_order N] : partial_order (lex (α →₀ N)) :=
+partial_order.lift _ (@id (function.injective (_ : lex (α →₀ N) → lex (α → N)))
+  finsupp.coe_fn_injective)
+
+local attribute [instance] finsupp_partial_order
+
+/--  The linear order on `finsupp`s obtained by the lexicographic ordering. -/
+noncomputable def lex [linear_order N] :
+  linear_order (lex (α →₀ N)) :=
```

Comment:
```suggestion
noncomputable instance lex.linear_order [linear_order N] : linear_order (lex (α →₀ N)) :=
```

Reply:
Same comment as above: should this really be a global instance?

Reply:
Yes: because what other order would ou expect on `lex (α →₀ N)` other than the lexicographic one?